### ISSUE TITLE
Intermittent Finalization Failure

### ISSUE THREAD

**Issue by bxpana** [matter-labs/dapp-portal#193]
As shown in https://github.com/zkSync-Community-Hub/zksync-developers/discussions/719, https://github.com/zkSync-Community-Hub/zksync-developers/discussions/711, and https://github.com/zkSync-Community-Hub/zksync-developers/discussions/636 there are intermittent issues where finalizing and claiming withdrawals from https://portal.zksync.io/ does not work as expected and results in the error message:
> Error Message:
Fee estimation error: The contract function "finalizeWithdrawal" reverted with the following reason:
xx

**Comment by popzxc** [comment]
Error originates from `_proveL2LogInclusion` function:

```
    /// @dev Prove that a specific L2 log was sent in a specific L2 batch number
    function _proveL2LogInclusion(
        uint256 _batchNumber,
        uint256 _index,
        L2Log memory _log,
        bytes32[] calldata _proof
    ) internal view returns (bool) {
        require(_batchNumber <= s.totalBatchesExecuted, "xx");
      // ...
```

I guess the contract is invoked with invalid parameters under some circumstances.

**Comment by popzxc** [comment]
I also see that we have [something like this](https://github.com/matter-labs/dapp-portal/blob/3884936578320da4a1c26040b68db58861f7d825/utils/formatters.ts#L130) to report that withdrawal is already finalized, but:
1. According to the reports, I guess it's not true?
2. Looking at reported error messages, looks like now there is a newline in the error, so this handler probably doesn't work.

**Comment by popzxc** [comment]
Hopefully, this has been fixed by #197. If not, let's reopen.
